NitroxClient/MonoBehaviours/Debugging/SceneDebugger.cs

```diff
@@ -0,0 +1,253 @@
+                                        float.TryParse(GUILayout.TextField(pos.y.ToString()), NumberStyles.Float, CultureInfo.InvariantCulture, out pos.y);
+                                        float.TryParse(GUILayout.TextField(pos.z.ToString()), NumberStyles.Float, CultureInfo.InvariantCulture, out pos.z);
+                                        selectedObject.transform.position = pos;
+                                    }
+                                }
+
+                                //Add other component interfaces.
+                                foreach (MonoBehaviour behaviour in selectedObject.GetComponents<MonoBehaviour>())
+                                {
+                                    Type script = behaviour.GetType();
+                                    using (new GUILayout.VerticalScope("Box"))
+                                    {
+                                        GUILayout.Label(script.Name);
+                                    }
+                                }
+                            }
+                        }
+                        else
+                        {
+                            GUILayout.Label($"No selected GameObject\nClick on an object in '{Tabs[1]}'", "fillMessage");
+                        }
+                    }
+                    break;
+            }
+        }
+
+        protected override void OnSetSkin(GUISkin skin)
+        {
+            base.OnSetSkin(skin);
+
+            skin.SetCustomStyle("sceneLoaded", skin.label, s =>
+            {
+                s.normal = new GUIStyleState()
+                {
+                    textColor = Color.green
+                };
+                s.fontStyle = FontStyle.Bold;
+            });
+
+            skin.SetCustomStyle("loadScene", skin.button, s =>
+            {
+                s.fixedWidth = 60;
+            });
+
+            skin.SetCustomStyle("mainScene", skin.label, s =>
+            {
+                s.margin = new RectOffset(2, 10, 10, 10);
+                s.alignment = TextAnchor.MiddleLeft;
+                s.fontSize = 20;
+            });
+
+            skin.SetCustomStyle("fillMessage", skin.label, s =>
+            {
+                s.stretchWidth = true;
+                s.stretchHeight = true;
+                s.fontSize = 24;
+                s.alignment = TextAnchor.MiddleCenter;
+                s.fontStyle = FontStyle.Italic;
+            });
+
+            skin.SetCustomStyle("breadcrum", skin.label, s =>
+            {
+                s.fontSize = 20;
+                s.fontStyle = FontStyle.Bold;
+            });
+
+            skin.SetCustomStyle("breadcumNav", skin.box, s =>
+            {
+                s.stretchWidth = false;
+                s.fixedWidth = 100;
+            });
+        }
+
+        private void GuiRenderTree(IEnumerable<Transform> objects, int level = 0)
```

Comment:
This function doesn't seem to be used anywhere. Rather, it seems to be somewhat implemented when rendering `ActiveTab` `1`.

Reply:
Good catch. I first hard Resharper warn me about many things like this but I don't have a license for it anymore. You got recommendations for Visual Studio 2017 extensions and tools?

Reply:
I don't sadly. You should be able to generate some offline reports, though in this case I just noticed because I wanted to know where this was used.